### DISCUSSION TITLE
showing Youtube Iframe inside twig output

### DISCUSSION THREAD

**Discussion by Yrerbal** [symfony/symfony#43105] [Q&A]
Hello everyone,
I would like to ask a little question here, I have a misbehave on an twig output.
my issue is that i want to display a youtube iframe inside a classic String field, (articleContent in my case) but when i reach the linked article page, i always end up with a black square where the iframe should have been saying that "video unavailable | watch on youtube"
I think that i'm using the raw filter right, so i tried to print the iframe out inside another field, but the issues is still here. did anyone have a nice solution so i can let y users adding  youtube iframe nicely and simply as a "string" component of my articles's content ?
```
<div class="col-md-10 col-lg-8 col-xl-7 pt-10 container-fluid">
    {{article.texte|raw|raw}}
    {{article.iframetry|raw}}
    <h6 style="color: #800707; font-style: italic ">....
//////////Iframe :
<iframe id="ytplayer" type="text/html" width="720" height="405"
src="https://www.youtube.com/embed/YqvCptqhHfs?enablejsapi=1&modestbranding=1"
frameborder="0" allowfullscreen> 
```
output :
![image](https://user-images.githubusercontent.com/11061203/134013547-a9e50b22-87e0-4d1b-bdfa-892990c190b3.png)

(the upper part is the article.text|raw|raw, and the bottom one is the direct try only with the iframe in it)
thanks every one, especialy sensio labs teams ! cheers up and have a nice week

**Comment by stof** [comment]
first, doubling the `|raw` filter is totally useless. the effect of the `|raw` filtering is only to mark the value as safe for the auto-escaper. Doing that multiple times is useless.

regarding the iframe HTML being escaped, can you add a `{% dump article.iframetry %}` in your template and then look at the Debug panel of the web debug toolbar ? I'm wondering whether `article.iframetry` contains the expected HTML or whether the `<` are already encoded in it (in which case the bug is in the place doing that encoding, not in this template)

**Reply by Yrerbal** [reply]
Hello Stof, and thanks for your prompt reply, 

Exact ! i did add a second raw filter when i was making my template, i might had a trouble with my navigator cache, because i remember getting the thing working only when i added the second raw, but after testing it again right now, you are right, so thank's for this one. 

for the dump, everything seems to be quiet normal for me here is the result 
![image](https://user-images.githubusercontent.com/11061203/134031151-93e1f08a-fbee-444b-9992-c8e73d263c71.png)

**Comment by mathiasstocker** [comment]
As I also mentioned in the symfony support slack channel to your question:
IMHO this is not symfony / twig related. 
Are you sure the publisher of the video allows embedding?

If i click on the "Watch on YouTube" link of your example iframe, hit the "share" button and copy the embed code to a simple static html file i also get the same "Video unavailable" message.

So i guess embedding is disabled by the publisher.

A YT not Symfony related problem.

**Reply by Yrerbal** [reply]
Hello mathiasstocker, 

thanks a lots for involving into my question, i didn't noticed that eather, i'm gonna test it right now. 

hum... it seems you are right, here is what i got now : so it seems thats iframe do not longer appear as a "safe" web browser option....  here is what i got : 
![image](https://user-images.githubusercontent.com/11061203/134036594-72aff8aa-7f35-4c37-af3d-f29562a7fbe1.png)

so it seems that now i have to alter my web browser behave to allow displaying iframe ??

**Reply by stof** [reply]
well, looks like this www.youtube.com URL redirected to consent.youtube.com instead of showing the embed, and that other domain is not embeddable. I suggest you report it to youtube.

In any case, that's not a Symfony or Twig issue (which are only responsible for generating the HTML, which is probably correct if you view the page source).

**Reply by Yrerbal** [reply]
Hello stof, 
just to update that I finaly find my way out using the youtube-nocookie tag :  https://youtube-nocookie.com/embed/VIDEOID, if anything happens on this tag, i'll get my issue again so i'll report it to youtube, 
thanks !
